api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/CommonQueries.java

```diff
@@ -148,4 +150,41 @@ public static String getPatientsOnTbTreatmentQuery(
 
     return replaced;
   }
+
+  /**
+   * Gets last obs with value coded before enDate
+   *
+   * @param encounterTypeId The Obs encounter Type
+   * @param question The Obs quetion concept
+   * @param answers The third value coded
+   * @return String
+   */
+  public static String getLastCodedObsBeforeDate(
```

Comment:
@iteliolucas this should not be in `CommonQueries` if it is not specified in  [Common Queries](https://docs.google.com/document/d/1EtpeIn-6seD5skZJteCdANhxkKXQye9RckGV2eoYj6c/edit?usp=sharing).
Also what is the reason from re-implementing something that is already in `CodedObsCohortDefinition`?

Reply:
Already moved the query into `GenericCohortQueries`.

The reason for re-implementig part of with `CodedObsCohortDefinition` is that we are not 100% sure about the way `CodedObsCohortDefinition` behaves if there is a typo or any miss writing, sometimes it won't return the correct output. This was the case with DSD N3.